### ISSUE TITLE
After compile not visible on LAN

### ISSUE THREAD

**Issue by Rick4hun** [AuxXxilium/arc#409]
Compile many more model no, inter/amd cpu, same result, tried with internal i226v double nic or usb2lan r8152 adapter, can't find on LAN. It's a Topton N1 pro mini pc, with  amd ryzen 5 5500u, 16GB DDR4 sodimm, etc etc. Tried put all modules, addons, tried without, no luck.

**Comment by AuxXxilium** [comment]
look wiki for bios settings

**Comment by Rick4hun** [comment]
Thank you for your reply, tried disable all from wiki, but doesn't affected, then tried disabled step by step everything, doesn't help, get ip, compile loader, reboot, get ip, but not visible. It's a topton n1 pro nas server.  Pulled two nvme ssd, wifi card (under dsm unusable, etc etc, but same result. With default bios settings rockstor, truenas, win 11  works good.

**Comment by AuxXxilium** [comment]
Screenshot from Arc - Sysinfo like mentioned everywhere if i a bug is found.

**Comment by Rick4hun** [comment]
![IMG_20230719_213711](https://github.com/AuxXxilium/arc/assets/139920348/c705605e-e133-4f26-ae49-356de2f72fbb)
![IMG_20230719_213730](https://github.com/AuxXxilium/arc/assets/139920348/af55f9e1-7152-4542-8d53-26575457ab9f)

Just tried arpl, arc, srpl n18i, tcrp, it can't boot. Tried with usb flash driver (more) sd cars (mini pc has a microSD slot) dedicated 60GB sata SSD, internal double 2.5G nic, usb2lan r8152 adapter, don't visible on lan. Try different methods 2 weeks ago day to day 6-8 hours daily... I hope your image will work once, because this is a best options- addons- interface.  Thanks!

**Comment by AuxXxilium** [comment]
Try directboot

**Comment by Rick4hun** [comment]
Tried DS1621 model, no arc patch, no addons, only few modules, with direct boot, samesttory, not visible. Tried with an msi ATX board and a ryzen 5 4500cpu, same method with all adons, without direct boot, and that hardware booted immediately. So i think, the issue will be  something with this mini pc hardware. But very strange, truenas, rockstor nas server softwares works good, xpenology not.

**Comment by AuxXxilium** [comment]
modules all, addons only the stock and direct boot on. all addons is more than wrong

**Comment by Rick4hun** [comment]
I completely understand, if my mini pc is an exception, but don't understand, what part of hardware complete block the using, because it's a standard pc with an x64 cpu, real pci express slots, memory modules, sata and nvme  HDDs and standart input- outputs.

**Comment by naeemsufian** [comment]
Same problem, after building loader no ip issued. if i manually edit config file using real mac address i works perfect. auto generated mac address not getting ip. i am using e1000e network card.

**Comment by AuxXxilium** [comment]
> Same problem, after building loader no ip issued. if i manually edit config file using real mac address it works perfect. auto generated mac address not getting ip. i am using HP Z240 SFF Workstation with e1000e network card.

hp need directboot. anything else will not work.

**Comment by naeemsufian** [comment]
> > Same problem, after building loader no ip issued. if i manually edit config file using real mac address it works perfect. auto generated mac address not getting ip. i am using HP Z240 SFF Workstation with e1000e network card.
> 
> hp need directboot. anything else will not work.

Yes It works now using directboot enabled.... thanks

**Comment by AuxXxilium** [comment]
> I completely understand, if my mini pc is an exception, but don't understand, what part of hardware complete block the using, because it's a standard pc with an x64 cpu, real pci express slots, memory modules, sata and nvme HDDs and standart input- outputs. I am wondering about, because truenas working on the fly, rockstor and win 11 working on the fly. I know users, who tried it use to a beelink mini pc, intel NUC, etc etc any chopped hardware, but all working, never found any informations about something blocked using xpenology. I think it will be a strange bios, so sent message about the selling- producing company.

dsm is a closed os like macos.

**Comment by Rick4hun** [comment]
Yes, i can understand that, but tried over 10 years without same issue. Tried with 100+ different hardwares, all was ok, and after 10 yeary my opinion was dsm can runs everything. But  unforunately today i tired in trying, i am must waiting to other image, or another informations to use it with that topton n1pro server... 
Never give up, but at this point i think, i waste time without any luck. 
Thank you for your patience, youe work is a wonderful piece, i use it at my another server.

**Comment by AuxXxilium** [comment]
one more point, what kind of drives do you try to use? only nvme? because there are no sata drives connected to your controller

**Comment by Rick4hun** [comment]
No, i want to use an 1TB, and a  6TB WD purple for test, and if everything's ok,  want to push into a 8TB WD red plus, but loader can boot without HDDs, tried with a ryzen5 4500+ Gigabyte ATX board, it booted fast, and then shows, i haven't attached disk (attached picture about that, it's shows on hungarian language. 
But that ATX board booted, worked, so i think, at this point doesn't affected attached disk, or diskess booting. 
I prepared 2x 512GBKingston nvme disks to r/w cache (this is the reason i preferred DS923+ image), but not necessary, tried compile and boot  with SSDs, but tried without, same result doesn't affected. 
![IMG_20230719_144530](https://github.com/AuxXxilium/arc/assets/139920348/7b31b4fd-4f02-4f93-a6bf-40cecc8dfbd2)

**Comment by AuxXxilium** [comment]
> No, i want to use an 1TB, and a 6TB WD purple for test, and if everything's ok, want to push into a 8TB WD red plus, but loader can boot without HDDs, tried with a ryzen5 4500+ Gigabyte ATX board, it booted fast, and then shows, i haven't attached disk (attached picture about that, it's shows on hungarian language) But that ATX board booted, worked, so i think, at this point doesn't affected attached disk, or diskess booting. I prepared 2x 512GBKingston nvme disks to r/w cache (this is the reason i preferred DS923+ image), but not necessary, tried compile and boot with SSDs, but tried without, same result doesn't affected. ![IMG_20230719_144530](https://user-images.githubusercontent.com/139920348/254947418-7b31b4fd-4f02-4f93-a6bf-40cecc8dfbd2.jpg)

first i can't read this message and second it will boot without hdd but you will not be able to install dsm without hdd. only nvme will not work. regardless what model you select.

**Comment by Rick4hun** [comment]
If mine can't boot, nobody cares, i can, or can't install it. :-)) Nothing booted, so at this moment i am far far away the installig step. :-)) 
After it can boot, i will restart, checking again a proper booting, and then  push a hdd, but i thinking about last two weeks sequence, swithing on, wait 3 minutes, and switching off ater an unsuccesful boot, and i think, my HDDs, have a small tolerance about that repeating on- off switching. I tried to boot after 2 weeks, and spent for that 6-8-10 hours daily, so i am over 2000 switching, without proper boot i can long  push power button only.  
I don't want to ruin my nearly brand new hdd.

**Comment by Rick4hun** [comment]
After many more unsuccesful trying, and waster hours, i am plan to working another project, so wat want to you decide about my opened issue? If remailns open, and you will build new version, i will try it, but in this case completely understand, want to close it. Thank you for your work and your  advice.
